test/Microsoft.AspNet.Mvc.Core.Test/Rendering/HtmlHelperCheckboxTest.cs

```diff
@@ -214,16 +214,46 @@ public void CheckBoxWithComplexExpressionsEvaluatesValuesInViewDataDictionary()
         }
 
         [Fact]
-        public void CheckBoxForWithInvalidBooleanThrows()
+        public void CheckBoxForWithNullContainer_TreatsBooleanAsFalse()
         {
             // Arrange
-            var expected = "String was not recognized as a valid Boolean.";
-            var helper = DefaultTemplatesUtilities.GetHtmlHelper(GetTestModelViewData());
+            var expected = @"<input id=""Property1"" name=""Property1"" type=""checkbox"" value=""true"" />" +
+                           @"<input name=""Property1"" type=""hidden"" value=""false"" />";
+            var viewData = GetTestModelViewData();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(viewData);
+            var valueProviderResult = new ValueProviderResult("false", "false", CultureInfo.InvariantCulture);
+            viewData.ModelState.SetModelValue("Property1", valueProviderResult);
 
-            // Act & Assert
-            // "Property2" in ViewData isn't a valid boolean
-            var ex = Assert.Throws<FormatException>(() => helper.CheckBoxFor(m => m.Property2));
```

Comment:
the new test in this slot covers the same scenario but with updated expectations.  if `CheckBoxFor` ignores `ViewData` (as it should), it's not possible to have a conversion issue here.  the `CheckBox` version of this test still `throw`s however.

side note: code does conversions when grabbing from `ModelState` but just ignores the attempted value if it doesn't convert.